torchvision/transforms/transforms.py

```diff
@@ -527,6 +527,69 @@ def __repr__(self):
         return self.__class__.__name__ + '(p={})'.format(self.p)
 
 
+class RandomPerspective(object):
+    """Performs Perspective transformation of the given PIL Image randomly with a given probability.
+
+    Args:
+        p (float): probability of the image being perspectively transformed. Default value is 0.5
+        interpolation : Default- Image.BICUBIC
+    """
+
+    def __init__(self, interpolation=Image.BICUBIC, p=0.5):
+        self.p = p
+        self.interpolation = interpolation
+
+    def __call__(self, img):
+        """
+        Args:
+            img (PIL Image): Image to be Perspectively transformed.
+
+        Returns:
+            PIL Image: Random perspectivley transformed image.
+        """
+        if not F._is_pil_image(img):
+            raise TypeError('img should be PIL Image. Got {}'.format(type(img)))
+
+        if random.random() < self.p:
+            width, height = img.size
+            coeffs = self.get_params(width, height)
+            return F.perspective(img, coeffs, self.interpolation)
+        return img
+
+    @staticmethod
+    def get_params(width, height):
+        """Get parameters for ``perspective`` for a random perspective transform.
+
+        Args:
+            width : width of the image.
+            height : height of the image.
+        Returns:
+            tuple: 8-tuple (a, b, c, d, e, f, g, h) which contains the coefficients
+                    for a perspective transform.
+        """
+        half_height = int(height / 2)
+        half_width = int(width / 2)
+        topleft = (random.randint(0, half_width), random.randint(0, half_height))
+        topright = (random.randint(half_width, width - 1), random.randint(0, half_height))
+        botright = (random.randint(half_width, width), random.randint(half_height, height - 1))
+        botleft = (random.randint(0, half_width), random.randint(half_height, height - 1))
+        matrix = []
+        startpoints = [(0, 0), (width - 1, 0), (width - 1, height - 1), (0, height - 1)]
```

Comment:
I think this block of code, which computes the coefficients, could be made its own function, as it will be useful for people wanting to use the functional interface

Reply:
@fmassa , i think we can have a function in the functional interface which computes the coordinates 
corresponding to two  quadrilaterals given as input.
Then for the random transform we can just have the random vertices's calling the functional interface.